tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php

```diff
@@ -0,0 +1,279 @@
+         ],
+         'Check for invalid arguments 3' => [
+            'data'     => [null, null, -10],
+            'expected' => ['return' => false, 'message' => 'The application does not exists'],
+         ],
+         'Check for correct value'       => [
+            'data'     => [null, $package, $package->getID()],
+            'expected' => ['return' => true],
+         ],
+      ];
+   }
+
+   /**
+    * @dataProvider applyProvider
+    * @tags testApply
+    * @param array $data
+    * @param array $expected
+    */
+   public function testApply(array $data, array $expected) {
+      $policyData = null;
+      if (true === $expected['return']) {
+         $policyData = new \PluginFlyvemdmPolicy();
+         $policyData->getFromDBBySymbol('removeApp');
+      }
+      list($policy) = $this->createNewPolicyInstance($policyData);
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmFleet');
+      $mockInstance->getMockController()->getID = 1;
+      if (true == $expected['return']) {
+         // the fleet must exist if we expect a true action.
+         $mockInstance = $this->createFleetInDB();
+      }
+      $result = $policy->apply($mockInstance, $data[0], $data[1], $data[2]);
+      $this->boolean($result)->isEqualTo($expected['return']);
+      if (!$expected['return'] && isset($expected['message'])) {
+         $this->string($_SESSION["MESSAGE_AFTER_REDIRECT"][1][0])->isEqualTo($expected['message']);
+      }
+      unset($_SESSION["MESSAGE_AFTER_REDIRECT"]); // to clear the buffer
+   }
+
+   /**
+    * @tags testGetMqttMessage
+    */
+   public function testGetMqttMessage() {
+      list($policy) = $this->createNewPolicyInstance();
+      $this->boolean($policy->getMqttMessage(null, null, null))->isFalse();
+
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmPackage');
+      $mockInstance->getMockController()->getID = '1';
+      $mockInstance->getMockController()->getFromDB = true;
+      $mockInstance->getMockController()->getField = 'myFakePackage';
+
+      $result = $policy->getMqttMessage('', $mockInstance, $mockInstance->getID());
+      $this->array($result)->hasKeys(['id', $this->dataField['symbol']])
+         ->string($result['id'])->isEqualTo($mockInstance->getID())
+         ->string($result[$this->dataField['symbol']])->isEqualTo($mockInstance->getField('package_name'));
+   }
+
+   /**
+    * @tags testShowValue
+    */
+   public function testShowValue() {
+      list($policy) = $this->createNewPolicyInstance();
+
+      // Invalid package ID
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmTask');
+      $mockInstance->getMockController()->getField = -1;
+      $result = $policy->showValue($mockInstance);
+      $this->string($result)->isEqualTo(NOT_AVAILABLE);
+   }
+
+   /**
+    * Create an application (directly in DB) because we are not uploading any file
+    * @return \PluginFlyvemdmPackage
+    */
+   private function createAppInDB() {
```

Comment:
This method should move in tests/src/Flyvemdm/TestingCommonTools.php . After we merge this PR, let's create a PR to share this method with other tests needing it.